File: server-nestjs-single/src/policies/CommentPolicy.ts
import { ResourcePolicy } from '@rhino-dev/rhino-nestjs';

/**
 * Comment policy — SINGLE-TENANT (role-free). Ownership inherited via
 * CommentScope (Task → Project userId). `userId` is the comment author, stamped
 * by StampProjectOwner on create and never client-trusted.
 */
export class CommentPolicy extends ResourcePolicy {
  override resourceSlug = 'comments';

  override permittedAttributesForShow(): string[] {
    return ['*'];
  }
  override hiddenAttributesForShow(): string[] {
    return [];
  }
  override permittedAttributesForCreate(): string[] {
    return ['userId', 'body', 'taskId'];
  }
  override permittedAttributesForUpdate(): string[] {
    return ['body'];
  }
}

File: server-rails-single/app/policies/comment_policy.rb
# frozen_string_literal: true

# Single-tenant CommentPolicy — permissive on attributes; ownership enforced by
# Scopes::CommentScope (comments under the current user's projects).
class CommentPolicy < Rhino::ResourcePolicy
  def permitted_attributes_for_show(_user) = ["*"]
  def hidden_attributes_for_show(_user) = []
  def permitted_attributes_for_create(_user) = %w[body task_id]
  def permitted_attributes_for_update(_user) = %w[body]
end
